docs/src/pages/components/masonry/ImageMasonry.tsx

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import * as React from 'react';
+import Masonry from '@material-ui/lab/Masonry';
+import MasonryItem from '@material-ui/lab/MasonryItem';
+
+export default function ImageMasonry() {
+  return (
+    <Masonry
+      columns={{ xs: 3, sm: 5 }}
+      spacing={{ xs: 1, sm: 1.5 }}
+      sx={{ height: 300 }}
+    >
+      {itemData.map((item) => (
+        <MasonryItem key={item.img}>
+          <img
+            srcSet={`${item.img}?w=164&h=164&auto=format 1x,
+            ${item.img}?w=164&h=164&auto=format&dpr=2 2x`}
```

Comment:
The image loading logic looks a LOT better. The only change I would make is to try to remove the height constraint, it seems to do nothing, w=164 wins over (it's not squared).

```suggestion
            srcSet={`${item.img}?w=164&auto=format 1x,
            ${item.img}?w=164&auto=format&dpr=2 2x`}
```

x the other cases